src/ext/resources/legacygraphqlapi/schema.graphqls

```diff
@@ -3694,6 +3893,30 @@ type TicketType implements Node {
     zones: [String!]
 }
 
+scalar TimeZone
+
+input TransferPreferences {
+    penalty: Int
+    minimumTime: Duration
+}
+
+input TransitModePreferences {
+    mode: TransitMode!
+    reluctance: Cost
+}
+
+input TransitPreferences {
+    """TODO should we reuse this or make a new design"""
+    unpreferred: InputUnpreferred
+    """TODO should we reuse this or make a new design"""
+    banned: InputBanned
+    board: BoardPreferences
+    alight: AlightPreferences
+    transfer: TransferPreferences
+    realtime: RealtimePreferences
+    modes: [TransitModePreference!]
```

Comment:
We should probably have a think how to integrate Bartosz's filter API here.

Reply:
True. I'm going to check at some point if I'm missing some other latest features from this API in this new draft or any cool features from the transmodel API.

Reply:
I haven't really studied the filter API closely but unless it has major issues, we can port it to this API as is.